### PR TITLE
[lunr] Rewritten for lunr.js v2.1.3

### PR DESCRIPTION
**Pull request by seantanly**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

----
The current definitions are outdated, meant for lunr.js < v2.0. 

This is a rewritten definition based on the current lunr.js v2.1.3.

### CODE REVIEW

**Comment by sebastian-lenz** [comment]
:thumbsup:

**Comment by seantanly** [comment]
@dt-bot @typescript-bot Please add to Pull Request Triage Backlog

**Comment by seantanly** [comment]
@sandersn 

can you help take a look what else is need to push this forward? 
@dt-bot and @typescript-bot aren't labelling or adding the PR to the [PR Burndown Board](https://github.com/DefinitelyTyped/DefinitelyTyped/projects/3?card_filter_query=is%3Aopen)

Thanks!

**Comment by sandersn** [comment]
@typescript-bot was out due to a build for a few days last week. Perhaps it just missed this PR? I'm not sure why.

Could you move the old files into a v1 directory? Here's [an explanation with a link to an example](https://github.com/DefinitelyTyped/DefinitelyTyped/#i-want-to-update-a-package-to-a-new-major-version)

**Comment by seantanly** [comment]
@sandersn

I have moved the old files to `v0` directory instead of `v1` because they were written based on `lunr v0.5.3` but encounters the following error on Travis CI. How do I fix that?

```console
 Error: lunr: Older version 0 must have a path mapping for itself.
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:260:9
    at Generator.next (<anonymous>)
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:7:71
    at Promise (<anonymous>)
    at __awaiter (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:3:12)
    at calculateDependencies (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:183:12)
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:99:47
    at Generator.next (<anonymous>)
    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/lib/definition-parser.js:4:58)
    at <anonymous>
npm ERR! Test failed.  See above for more details.
```

**Comment by sandersn** [comment]
Hm, @andy-ms is on vacation so I poked around the source. @RyanCavanaugh might be able to help too.

From what I can tell, if `oldMajorVersion` is defined (and it is -- `'0'` here), then the tsconfig must also have the `paths` option defined. Here's an example I pulled from the link I pasted above:

```json
        "paths": {
            "history": [ "history/v2" ]
        },
```

So I think you have to add the equivalent `"lunr": ["lunr/v0"]` in v0's tsconfig. (Also when I followed history's v2 to its source, there was another line `"history/*": ["history/v0/*"]`)

File: types/lunr/v0/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": false,
+        "strictNullChecks": false,
+        "baseUrl": "../../",
+        "typeRoots": [
+            "../../"
+        ],
+        "paths": {
+            "history": ["history/v0"]
```
Comment (sandersn) [inline types/lunr/v0/tsconfig.json]: This line needs to change 'history' to 'lunr'

**Comment by seantanly** [comment]
@sandersn thanks! The Travis CI tests passes now. I got confused about the example using `history`, thinking `history` is a keyword instead of package name. Thanks for clearing the confusion. :)

**Pull request merged**